odl/contrib/fom/supervised.py

```diff
@@ -353,189 +327,161 @@ def range_difference(data, ground_truth, mask=None, normalized=False):
-    (``smoothness_factor``) is a positive real number that controls the
-    'smoothness' of the weighting function :math:`\\alpha`.
+        \\alpha(x) = e^{-\\frac{1}{k} \\beta_m(x)},
 
-    The normalized FOM takes values in [0, 1], with higher correspondance
-    at lower FOM value.
-
-    Parameters
-    ----------
-    data : `FnBaseVector`
-        Input data or reconstruction.
-    ground_truth : `FnBaseVector`
-        Reference to compare 'data' to.
-    mask : `FnBaseVector`, optional
-        Binary mask to define ROI in which FOM evaluation is performed.
-    normalized  : bool, optional
-        Boolean flag to switch between unormalized and normalized FOM.
-    smoothness_factor : float, optional
-        Positive real number. Higher values gives smoother weighting.
-
-    Returns
-    -------
-    fom : float
-        Scalar (float) indicating weighted mean squared error between
-        ``data`` and ``ground_truth``. In normalized form the FOM takes
-        values in [0, 1], with higher correspondance at lower FOM value.
+    where :math:`\\beta_m(x)` is the Euclidian distance transform of a
+    given binary mask :math:`m`, and :math:`k` positive real number that
+    controls the smoothness of the weighting function :math:`\\alpha`.
+    The weighting gives higher values to structures outside the region
+    of interest defined by the mask.
     """
-    import scipy.ndimage.morphology as scimorph
+    from scipy.ndimage.morphology import distance_transform_edt
 
     if smoothness_factor is None:
         smoothness_factor = np.mean(data.space.shape) / 10
 
     if mask is not None:
-        mask = scimorph.distance_transform_edt(1 - mask)
+        mask = distance_transform_edt(1 - mask)
         mask = np.exp(mask / smoothness_factor)
 
-    fom = mean_squared_error(data,
-                             ground_truth,
-                             mask=mask,
-                             normalized=normalized)
-
-    return fom
+    return mean_squared_error(data, ground_truth, mask, normalized)
 
 
 def ssim(data, ground_truth,
-         size=11, sigma=1.5, K1=0.01, K2=0.03, dynamic_range=None,
-         normalized=False):
+         size=11, sigma=1.5, K1=0.01, K2=0.03, dynamic_range=None):
     """Structural SIMilarity between ``data`` and ``ground_truth``.
 
-    Evaluates `structural similarity
-    <https://en.wikipedia.org/wiki/Structural_similarity>`_ between
-    input (``data``) and reference (``ground_truth``).
+    The SSIM takes value -1 for maximum dissimilarity and +1 for maximum
+    similarity.
+
+    See also `this Wikipedia article
+    <https://en.wikipedia.org/wiki/Structural_similarity>`_.
 
     Parameters
     ----------
     data : `FnBaseVector`
-        Input data or reconstruction.
+        Input data to compare to the ground truth.
     ground_truth : `FnBaseVector`
         Reference to compare ``data`` to.
-    normalized : bool
-        TEXT
+    size :
+        TODO...
```

Comment:
Yupp

Reply:
Well, I already did more than necessary, but this would require me to check and see what the parameters do. Nope. This is on the original authors. Ping @davlars @zickert

Reply:
Leave it here then

Reply:
I added the two obvious ones, but the `K1`, `K2` and `dynamic_range` params are for later.

Reply:
To finish this could perhaps be a topic for the upcoming team workshop day.